base/version_comparable.go

```diff
@@ -49,31 +47,35 @@ func NewComparableVersionFromString(version string) (*ComparableVersion, error)
 	if err != nil {
 		return nil, err
 	}
-	return &ComparableVersion{
+	v := &ComparableVersion{
 		epoch:   epoch,
 		major:   major,
 		minor:   minor,
 		patch:   patch,
 		other:   other,
 		build:   build,
 		edition: edition,
-	}, nil
+	}
+	v.str = v.formatComparableVersion()
```

Comment:
Are you thinking this is safer than v.str=version?  If so, should version != v.formatComparableVersion() return error?

Reply:
I think that validation is good to do, I was pretty much preserving the behavior after removing the `sync.Once`

Reply:
You can end in situations where these aren't the same "1.0.0.0" will resolve to "1.0.0" for example. I don't think this is likely in practice.